product_kits_inventory/product.py

```diff
@@ -0,0 +1,313 @@
+from openerp.osv import orm, fields, osv
+from openerp.tools.translate import _
+import openerp.addons.decimal_precision as dp
+
+
+class ProductProduct(orm.Model):
+    _inherit = 'product.product'
+
+    def _kits_product_available(self, cr, uid, ids, field_names=None,
+                                arg=False, context=None):
+        res = {}
+        field_names = field_names or []
+        context = context or {}
+        for id in ids:
+            res[id] = {}.fromkeys(field_names, 0.0)
+        field_map = {
+            'kits_qty_available': 'qty_available',
+            'kits_incoming_qty': 'incoming_qty',
+        }
+        for product_record in self.browse(cr, uid, ids, context=context):
+            # check if is a kit product.
+            so_qty = self._get_sale_quotation_qty(cr, uid, product_record.id,
+                                                  context=context)
+            if not self._is_kit(
+                cr, uid,
+                [product_record.id],
+                    context=context).get(product_record.id):
+
+                res[product_record.id] = {
+                    'kits_qty_available': 0,
+                    'kits_incoming_qty': 0,
+                    'kits_sale_quotation_qty': so_qty
+                }
+            else:
+                flag = True
+                for bom in product_record.bom_ids:
+                    """ now get always get the default_for_kit of bom is \
+                        True and bom type is phantom."""
+                    if bom.default_for_kit and bom.type == 'phantom':
+                        child_product_res = {}
+                        for line in bom.bom_lines:
+                            child_product_res[line.product_id.id] = {
+                                'product_qty': line.product_qty or 0.0,
+                                'product_uom': line.product_uom,
+                                'uom': bom.product_id.uom_id,
+                                'mrp_product_qty': line.bom_id.product_qty,
+                                'mrp_product_uom': line.bom_id.product_uom}
+                        child_product_qtys = self._product_available(
+                            cr, uid, child_product_res.keys(),
+                            field_map.values(), context=context)
+                        res[product_record.id] = {
+                            'kits_qty_available': self._get_qty_from_children(
+                                cr, uid, child_product_qtys, child_product_res,
+                                'qty_available'),
+                            'kits_incoming_qty': self._get_qty_from_children(
+                                cr, uid, child_product_qtys, child_product_res,
+                                'incoming_qty'),
+                            'kits_sale_quotation_qty': so_qty
+                        }
+                        flag = False
+                        break
+                if flag:
```

Comment:
Instead of your flag, you can use the 'for…else' statement : https://docs.python.org/2/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops